nucleus/payara-modules/requesttracing-core/src/main/java/fish/payara/nucleus/requesttracing/store/ClusteredRequestTraceStore.java

```diff
@@ -109,6 +93,9 @@ public RequestTrace addTrace(RequestTrace trace, RequestTrace traceToRemove) {
 
     @Override
     public void setSize(int maxSize) {
+        while (store.size() > maxSize) {
+            store.remove(strategy.getTraceForRemoval(getTraces(), maxSize, null).getTraceId());
+        }
```

Comment:
The semantics of changing size seemed unclear and contradictory to the goal of never having more items then the `maxSize` so I decided to remove items using the strategy as long as the store has too many items for the new size.